src/lib/core/platform/browser.ts

```diff
@@ -0,0 +1,4 @@
+/** Gets whether the current platform is a web browser (versus node). */
+export function isBrowser() {
+  return typeof document === 'object';
```

Comment:
A couple of things:
* Technically shouldn't be an issue, but `typeof document === 'object'` will be true if `document` is `null`. 
* Perhaps this should be a `const`, instead of a function?

Also I was doing these checks inline in a few places, I don't know whether we shouldn't switch them to use `isBrowser` with this PR or in a follow-up.